packages/cosmic-swingset/src/block-manager.js

```diff
@@ -6,16 +6,52 @@ import { assert, details as X } from '@agoric/assert';
 
 import * as BRIDGE_ID from '@agoric/vats/src/bridge-ids.js';
 
+import { makeStoredSubscription, makeSubscriptionKit } from '@agoric/notifier';
 import * as ActionType from './action-types.js';
 import { parseParams } from './params.js';
 
+import '@agoric/notifier/exported.js';
+
 const console = anylogger('block-manager');
 
 // Artificially create load if set.
 const END_BLOCK_SPIN_MS = process.env.END_BLOCK_SPIN_MS
   ? parseInt(process.env.END_BLOCK_SPIN_MS, 10)
   : 0;
 
+/** @typedef {Record<string, unknown>} InstallationNotification */
+
+/**
+ * @template T
+ * @param {T} initialValue
+ * @param {StorageNode} [storageNode]
+ */
+const makeStoredPublisher = (initialValue, storageNode) => {
+  let providedPublication;
+
+  const provide = () => {
+    if (providedPublication) {
+      return providedPublication;
+    }
+    /** @type {SubscriptionRecord<T>} */
+    const { publication, subscription } = makeSubscriptionKit(initialValue);
```

Comment:
This should probably build on #5435 to avoid the cost of holding all results.
```suggestion
    const { publisher, subscriber } = makePublishKit(initialValue);
```